auth/credentials/clientcredentials.go

```diff
@@ -0,0 +1,78 @@
+package credentials
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/clientcredentials"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+// Config describes a 2-legged OAuth2 flow, with both the
+// client application information and the server's endpoint URLs.
+type Config struct {
+	clientcredentials.Config
```

Comment:
Do you have strong opinion on using config?
I'm concerned that this produces end user api that will expose too much.
Some examples:
```
	        // Token URL should be inherited from SDK. For all users will be hardcoded and not exposed to users.
		TokenURL:     "https://cloud.mongodb.com/oauth/token",
		// Scopes are present but they are not supported by our backend
		Scopes:       []string{"your-scope"},
            // By default will make redundant wrong requests against our API.
	      AuthStyle: oauth2.Auto
```

Reply:
yes it has the implementation to get tokens, if we don't use it you have to implement them which was the whole point of this change, not re-implement oauth

to your concerns there's NewConfig which encapsulates this logic to create the one you want, so following your comments of "users won't use this package directly, then they won't use this struct directly right?

on top of that for this one 
>// By default will make redundant wrong requests against our API.

on my tests I haven't' seen this to be true

Reply:
> if we don't use it you have to implement them which was the whole point of this change, not re-implement oauth

Prior to last commit, all changes we had reused `oauth.TokenSource.Token()` but also incorporated additional requirements from the scope: 
- Ability to integrate with Atlas CLI by creating transport instead of Client.
- Token being string. 
- Custom user agent etc. 
- RevokableTokenSource that provided immutable access to Token and Revoke so it can be part of the context without security risks.   

I like how lean and clean current approach is but I might need to reintroduce some removed code in order to support those features. Is that ok?

> on my tests I haven't' seen this to be true

We have provided override when creating config, but we are still exposing that field to end users which is my point.

Reply:
I don't see any of the points you mention captured in the scope

Reply:
Covered offline. We will settle with oauth2.Token as representation of token. Apply custom user agent.
We do not need RevocableTokenSource as plan is to contribute those changes to auth2 in long run.